functions.php

```diff
@@ -1385,90 +1385,91 @@ function wp_autoupdates_disable_auto_updates() {
-			foreach ( $wp_autoupdate_themes as $index => $theme_file ) {
-				if ( $theme_file === $asset ) {
-					unset( $wp_autoupdate_themes[ $index ] );
-					update_site_option( 'wp_auto_update_themes', $wp_autoupdate_themes );
-
-					/** This filter is documented in wp-admin/includes/class-wp-plugins-list-table.php */
-					$all_themes           = wp_get_themes();
-					$wp_autoupdate_themes = array_intersect( $wp_autoupdate_themes, array_keys( $all_themes ) );
-					$enabled_count        = count( $wp_autoupdate_themes );
-					$theme_name           = $all_themes[ $asset ]->get( 'Name' );
-
-					$theme_url = add_query_arg(
-						array(
-							'action'   => 'autoupdate',
-							'theme'    => $asset,
-							'_wpnonce' => wp_create_nonce( 'autoupdate-theme_' . $asset ),
-						),
-						'themes.php'
-					);
-
-					$return_html = sprintf(
-						'<p class="theme-autoupdate-disabled"><a href="%s" class="edit theme-autoupdate-enable" aria-label="%s"><span class="theme-autoupdate-label">%s</span></a></p>',
-						esc_url_raw( $theme_url ),
-						esc_html__( 'Enable automatic updates for', 'wp-autoupdate' ) . ' ' . esc_html( $theme_name ),
-						esc_html__( 'Enable auto-updates', 'wp-autoupdates' )
-					);
-
-					wp_send_json_success(
-						array(
-							'enabled_count'  => '(' . absint( $enabled_count ) . ')',
-							'disabled_count' => '(' . absint( count( $all_themes ) - $enabled_count ) . ')',
-							'return_html'    => wp_kses_post( $return_html ),
-							'type'           => 'theme',
-						)
-					);
-					break;
-				}
-			}
-		}
+		$wp_autoupdate_themes = (array) get_site_option( 'wp_auto_update_themes', array() );
+		$wp_autoupdate_themes = array_diff( $wp_autoupdate_themes, array( $asset ) );
+		update_site_option( 'wp_auto_update_themes', $wp_autoupdate_themes );
+
+		/** This filter is documented in wp-admin/includes/class-wp-plugins-list-table.php */
+		$all_themes           = wp_get_themes();
+		$wp_autoupdate_themes = array_intersect( $wp_autoupdate_themes, array_keys( $all_themes ) );
+		$enabled_count        = count( $wp_autoupdate_themes );
+
+		$theme_url = add_query_arg(
+			array(
+				'action'   => 'autoupdate',
+				'theme'    => $asset,
+				'_wpnonce' => wp_create_nonce( 'autoupdate-theme_' . $asset ),
+			),
+			'themes.php'
+		);
+
+		$aria_label = esc_attr(
+			sprintf(
+				/* translators: Theme name. */
+				_x( 'Enable automatic updates for %s', 'plugin', 'wp-autoupdates' ),
+				esc_html( $all_themes[ $asset ]->get( 'Name' ) )
+			)
+		);
+
+		$return_html = sprintf(
+			'<p class="theme-autoupdate-disabled"><a href="%s" class="edit theme-autoupdate-enable" aria-label="%s"><span class="theme-autoupdate-label">%s</span></a></p>',
+			esc_url_raw( $theme_url ),
+			$aria_label,
+			esc_html__( 'Enable auto-updates', 'wp-autoupdates' )
+		);
+
+		wp_send_json_success(
+			array(
+				'enabled_count'  => '(' . absint( $enabled_count ) . ')',
```

Comment:
Is there a possibility for this count to be more than 1? As far as I see no, the bulk actions do not use AJAX. Then perhaps the counts should be updated from js, with `success` meaning `1` and `error` meaning no update.

Again, seems the proper place to fix this would be while fixing wp-qutoupdates.js.

Reply:
Yes, it can be greater than 1: `$enabled_count`  is the count of plugins/themes that have auto-updates enabled **after** this one plugin/theme has had auto-updates enable/disabled and is used by the JS to update the counts in the `Views` above the list table, e.g.,

`All (60) | Active (17) | Inactive (43) | Auto-updates Enabled (19) | Auto-updates Disabled (41)`

Reply:
Uh, sorry I mean is there a chance for the difference to be more than 1. Not related for the fix here though, seems better to look at it when looking at the js.